src/parcsr_mv/par_csr_matrix.c

```diff
@@ -423,25 +423,34 @@ hypre_ParCSRMatrixSetNumRownnz( hypre_ParCSRMatrix *matrix )
 
    HYPRE_BigInt      local_num_rownnz;
    HYPRE_BigInt      global_num_rownnz;
-   HYPRE_Int         i, j;
+   HYPRE_Int         i = num_rownnz_diag;
+   HYPRE_Int         j = num_rownnz_offd;
 
    if (!matrix)
    {
       hypre_error_in_arg(1);
       return hypre_error_flag;
    }
 
-   local_num_rownnz = i = j = 0;
-   while (i < num_rownnz_diag && j < num_rownnz_offd)
+   local_num_rownnz = 0;
+   if (rownnz_diag && !rownnz_offd)
    {
-      local_num_rownnz++;
-      if (rownnz_diag[i] < rownnz_offd[j])
-      {
-         i++;
-      }
-      else
+      local_num_rownnz = num_rownnz_diag;
```

Comment:
`local_num_rownnz` is overwritten below (see line 457). Particularly, with your initialization of `i` and `j` above, I don't think it will end up with the right value?

Reply:
The idea here is that `(num_rownnz_diag - i) + (num_rownnz_offd - j)` is zero, so line 457 doesn't change `local_num_rownnz`